onnxruntime/core/providers/nuphar/scripts/model_quantizer.py

```diff
@@ -158,7 +158,7 @@ def quantize_matmul_2d_with_weight(in_node, in_graph, nf, converted_weights, qua
             Q_Xf = nf.make_node('Mul', [norm_X, np.asarray(x_qcfg.q_range()).astype(np.float32)])
             Q_Xf = nf.make_node('Add', [Q_Xf, np.asarray(0.5).astype(np.float32)])
             Q_Xf = nf.make_node('Floor', Q_Xf)
-            Q_Xf = nf.make_node('Clip', Q_Xf, {'max':x_qcfg.q_max(), 'min':x_qcfg.q_min()})
+            Q_Xf = nf.make_node('Clip', [Q_Xf, np.asarray(x_qcfg.q_min()).astype(np.float32), np.asarray(x_qcfg.q_max()).astype(np.float32)])
```

Comment:
Can we add different versions of the Clip op based on the op version? For example, we could have something like:

    op_ver = ensure_opset(out_mp, 10)
    ...
    quantize_matmul_2d_with_weight(..., op_ver)

and then

    def quantize_matmul_2d_with_weight(..., op_ver):
        ...
        if op_ver > 10:
            Q_Xf = nf.make_node('Clip', [Q_Xf, np.asarray(x_qcfg.q_min()).astype(np.float32), np.asarray(x_qcfg.q_max()).astype(np.float32)])
        else:
            Q_Xf = nf.make_node('Clip', Q_Xf, {'max':x_qcfg.q_max(), 'min':x_qcfg.q_min()})

Reply:
Yes the editing needs to differentiate different opset versions.

---
In reply to: [424906962](https://github.com/microsoft/onnxruntime/pull/3939#discussion_r424906962) [](ancestors = 424906962)